src/typelib.erl

```diff
@@ -86,7 +86,9 @@ parse_type(Src) ->
 %% kept for user-defined types and record types. Filename is used to
 %% disambiguate between types with the same name from different modules.
 %% Annotated types as in Name :: Type are also removed.
--spec remove_pos(type()) -> type().
+-spec remove_pos(type() | [type()]) -> type().
+remove_pos(Types) when is_list(Types) ->
+    [ remove_pos(Ty) || Ty <- Types ];
```

Comment:
The spec is wrong. It's actually `([type()]) -> [type()]; (type()) -> type()`. I don't like this since it makes it more complex. Let it work on a single type only and let the caller do the comprehention (or lists:map).

Reply:
[What about `pp_type` then?](https://github.com/josefs/Gradualizer/commit/af616e05751f8d00e2f6d5a7a0a63ee1d04e6617#diff-24fdf073a8645c401774bfbe51791d5c0c5f4aaecf41856284451c8498fbe938R26-R32) If we don't want to consistently approach fun types like this, then I think the cleanup should be done as a separate refactoring ticket - possibly by changing fun representation, not just adjusting the call sites. This PR fixes a bug reported by an end user, so I'd like the fix to be available relatively promptly.

Reply:
Yeah it's a mess that we keep pushing around.

If we allow type to be a list in more places, it becomes even harder to get rid of it I think....

Reply:
A "bounded type" can also have guards, but types can't. That's another mess.